Quest-4/Part-2/main.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+def main():
+    fileName = "input.txt"
+    lines = readFile(fileName)
+
+    cards = getcards(lines)
+
+    for card in cards:
+        win = calculateWin(card["winNumbers"], card["cardNumbers"])
+        if win != 0:
+            for i in range(card["cardNumber"], card["cardNumber"] + win):
+                cards.append(cards[i])
```

Comment:
it works, though you could totally just scan original deck once, not adding to the end, but increasing counts of cards, multiplying win.